enhancements/compound-dependency-selectors.md

```diff
@@ -256,38 +227,96 @@ This is an example of a disjunction of conjunctions, where one, the other, or bo
 can be satisfy the constraint.
 
 ```yaml
-schema: olm.bundle
-name: baz.v1.0.0
-properties:
-- type: olm.constraint
-  value:
-    message: Required for Baz because...
-    any:
-      constraints:
-      - all:
-          constraints:
-          - package:
-              name: foo
-              versionRange: '>=1.0.0'
-          - gvk:
-              group: foos.example.com
-              version: v1
-              kind: Foo
-      - all:
-          constraints:
-          - package:
-              name: foo
-              versionRange: '<1.0.0'
-          - gvk:
-              group: foos.example.com
-              version: v1beta1
-              kind: Foo
+type: olm.constraint
+value:
+  failureMessage: Required for Baz because...
+  any:
+    constraints:
+    - all:
+        constraints:
+        - package:
+            name: foo
+            versionRange: '>=1.0.0'
+        - gvk:
+            group: foos.example.com
+            version: v1
+            kind: Foo
+    - all:
+        constraints:
+        - package:
+            name: foo
+            versionRange: '<1.0.0'
+        - gvk:
+            group: foos.example.com
+            version: v1beta1
+            kind: Foo
 ```
 
 The maximum raw size of an `olm.constraint` is 64KB to limit resource exhaustion attacks.
 See [this issue][json-limit-issue] for details on why size is limited and not depth.
 This limit can be changed at a later date if necessary.
 
+#### Negation
+
+Negation is worth further explanation, since at first glance its semantics
+are unclear in this context. The negation is really instructing the resolver
+to remove any possible solution that includes a particular GVK, package
+at a version, or satisfies some child compound constraint from the result set.
+
+If you define a negation constraint on a resource, then that resource cannot 
+be used as a solution. This can be useful if you want to have a constraint 
+that cannot be solved by a specific GVK or bundle. Most of the time, you will
+want to use this as a nested constraint. This is because the negation constraint
+in the root will not actually restrict anything. 
+
+**Remember, negation is only used to say "this package cannot be used as a 
+solution". It does not mean that the bundle is restricted from successfully 
+installing if the negated package is already installed on the cluster**
+
+The most common use case for negation is if you want to depend on a GVK but don't
+want a certain package/version of a package to provide it. In the below example, if the 
+solution were to include a package that would provide the `bufs.example.com/v1` GVK then 
+the `not` would remove it only if it was package `foo` with version `>=v1.0.0`. If no other 
+solutions are available then the bundle constraints would be considered unsolvable.
```

Comment:
```suggestion
The most common usecase for negation is excluding a certain package/version of a package
from providing a GVK that your operator depends on. In the below example, if the only package
providing the `bufs.example.com/v1` GVK was named `foo` with version `>=v1.0.0` it would
be removed from the solution and the constraints would be considered unsolvable.
```